src/main/java/pl/futurecollars/invoicing/service/IdService.java

```diff
@@ -0,0 +1,45 @@
+package pl.futurecollars.invoicing.service;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class IdService {
+
+  private static int id = 1;
+  private final Path idFilePath;
+
+  public IdService(Path idFilePath) {
+    this.idFilePath = idFilePath;
+    try {
+      File idFile = new File(String.valueOf(idFilePath));
+      if (!idFile.exists() || Files.readString(Paths.get(String.valueOf(idFilePath))).isEmpty()) {
+        idFile.createNewFile();
+        Files.writeString(idFilePath, String.valueOf(id));
+      }
+    } catch (IOException exception) {
+      System.out.println("Creation of idFile failed");
+      exception.printStackTrace();
+    }
+  }
+
+  public int getId() {
+    try {
+      String actualId = Files.readString(idFilePath);
```

Comment:
jeśli za każdym razem zapisujesz i odczytujesz id z pliku to nie potrzebujesz pola id.